### PR TITLE
Chapter 2: possible code improvements

### PR DESCRIPTION
**Pull request by pkourdis**
- Some possible code improvements:

- Maybe you can discuss also the more generic way to get only numerical or categorical attributes:

### CODE REVIEW

**Comment by ageron** [comment]
Thanks @pkourdis , these are nice improvements.  Could you please fix the minor URL issue that I mentioned in the comments, and I'll merge this PR.

**Comment by pkourdis** [comment]
I could not locate the comment for the minor URL issue. Could you please point out the comments? Thanks.

**Comment by ageron** [comment]
Hi @pkourdis,

Oops, it seems that I need to learn how reviews work in GitHub, sorry about that. It seems that I am the only one who can see this comment, for some reason (I cannot see it when I am not logged it).

I added a GitHub comment for the line:
```
-    "HOUSING_URL = DOWNLOAD_ROOT + \"datasets/housing/housing.tgz\"\n",
+    "HOUSING_URL = DOWNLOAD_ROOT + HOUSING_PATH + \"/housing.tgz\"\n",
```

My point was that `HOUSING_PATH` is a file path, not a URL.  On Windows it will contain backslashes. You don't want backslashes in a URL.

Thanks for your help!

File: 02_end_to_end_machine_learning_project.ipynb [hunk 1]
```diff
@@ -91,7 +93,7 @@
     "\n",
     "DOWNLOAD_ROOT = \"https://raw.githubusercontent.com/ageron/handson-ml/master/\"\n",
     "HOUSING_PATH = os.path.join(\"datasets\", \"housing\")\n",
-    "HOUSING_URL = DOWNLOAD_ROOT + \"datasets/housing/housing.tgz\"\n",
+    "HOUSING_URL = DOWNLOAD_ROOT + HOUSING_PATH + \"/housing.tgz\"\n",
```
[the remark this answers is not in the document]
Comment (pkourdis) [reply 02_end_to_end_machine_learning_project.ipynb]: @ageron you are right. I revert it back how you had it before.

**Comment by ageron** [comment]
Merged! Thanks a lot @pkourdis .

**Comment by pkourdis** [comment]
@ageron you are welcome!

**Pull request merged**